### PR TITLE
Make SDK search docs more clear

### PR DESCRIPTION
**Pull request by snide**
After chatting with @richardgill and a user on discord over the last day, I realized our search docs didn't do a great job explaining certain core features. Specifically the highlight examples weren't very clear.

Using Keanu Reeves is a good example for the typo tolerance examples, but using `Users` and `Posts` along with that term just confused the examples (especially when you're filtering posts by a `label` with a value of `movies` :dizzy_face:). I rewrote the tables to be `Movies` and `Actors`.

I also went and added line highlights to all the examples and fixed a lot of the code (which were referencing object columns :facepalm: ).

### CODE REVIEW

File: 040-SDK/111-search.mdx [hunk 1]
```diff
@@ -7,48 +7,55 @@ slug: sdk/search
 published: true
 ---
 
-As you insert data into Xata, it is automatically indexed for full-text search. You can run a search by using the [/search endpoint](/docs/api-reference/db/db_branch_name/tables/table_name/search#free-text-search-in-a-table). While the `/query` endpoint exists only at the table level, the `/search` endpoint exists both at the database branch and the table level, because it is possible to search across table.
+Xata natively provides text-based search for your database with the following core features:
 
-The search index is updated asynchronously after each insert/update, meaning that the search results are eventually consistent with the results that you get from the `/query` endpoint. Another fundamental difference to the `/query` endpoint is that `/search` doesn't support following links. This means that for links records you can filter by the ID of the linked record, but not any of the other columns. If you need filter or search by linked columns, other than the ID, it is recommended that you denormalize the data.
+- Fuzzy matching to account for typos.
```
Comment (joan-ing) [inline 040-SDK/111-search.mdx:12]: Would suggest parallelism for list 

- Fuzzy matching for typo tolerance in search queries
- Query limits to specified tables and columns
- Pre-search filters for searching within specific data subsets
- Weights and boosts to influence the relevance of search scores based on certain columns
- Highlight markup in search results to emphasize partial matches in returned content

Comment (snide) [reply 040-SDK/111-search.mdx:12]: > Query limits to specified tables and columns

A limit can mean something in programming, so this read as "Query limit". The parameter is also named `filter`, which is why that language was used. Reworded it to the below to match your sentence structure.

> Targets allow isolation of specific tables and columns in search queries

**Comment by snide** [comment]
@richardgill applied your feedback from Slack. Added you as reviewer.

**Review by richardgill — APPROVED** [review]
LGTM

**Pull request merged**